packages/client/src/hooks/useAccount.js

```diff
@@ -37,35 +42,70 @@ export default function useAccount() {
     }
   };
 
-  const getBalanceByCoinType = async (coinType, address) => {
-    const cadence = GetAccountBalanceByContractName(
-      COIN_TYPE_TO_META[coinType].contractName
-    );
-    const vaultPath = COIN_TYPE_TO_META[coinType].publicBalancePath;
-    const balance = await query({
-      cadence,
+  const getVaultExists = async (contractName, vaultPath, address) =>
+    query({
+      cadence: VAULT_EXIST_CHECK(contractName),
+      args: (arg, t) => [arg(address, t.Address), arg(vaultPath, t.Path)],
+    });
+
+  const getBalanceByCoinType = async (contractName, vaultPath, address) =>
+    query({
+      cadence: GET_ACCOUNT_BALANCE(contractName),
       args: (arg, t) => [arg(address, t.Address), arg(vaultPath, t.Path)],
     });
-    return balance;
-  };
 
   const getUserBalances = async (address) => {
     const result = COIN_TYPE_LIST.map((coinType) => ({
       coinType,
-      balance: 0,
     }));
     for await (const coin of result) {
+      const { coinType } = coin;
       try {
-        const coinBalance = await getBalanceByCoinType(coin.coinType, address);
-        coin.balance = coinBalance;
+        const contractName = COIN_TYPE_TO_META[coinType].contractName;
+        const vaultPath = COIN_TYPE_TO_META[coinType].publicBalancePath;
+        const vaultExists = await getVaultExists(
+          contractName,
+          vaultPath,
+          address
+        );
+        if (vaultExists) {
+          const coinBalance = await getBalanceByCoinType(
+            contractName,
+            vaultPath,
+            address
+          );
+          coin['balance'] = coinBalance;
+        }
       } catch (error) {
         console.log(`error getting balance for ${coin.coinType}`, error);
       }
     }
-    return result;
+    return result.filter((balance) => balance.balance);
+  };
+
+  const getUserVaults = async (address) => {
```

Comment:
do we need `getUserVaults` func? can we use `getUserBalances` and check if `balance is > 0` to know whether vault exist?

Reply:
Vault can exist with 0 balance, and it's difficult to differentiate if the vault exists or not if both return 0 as a result - but in our use case it's okay to do it in a way you proposed!

Reply:
On page where we see treasury vaults -> we need the information if we should enable/disable the button for depositing tokens, there we don't need info about balance, but rather if the vault exists or not on the user side -> so that way we avoid more requests than needed (the number of requests doubles if we call it on that page).

Reply:
make sense. thanks for explaining to me!